src/main/java/com/ctc/wstx/sr/StreamScanner.java

```diff
@@ -1186,57 +1186,69 @@ protected int resolveSimpleEntity(boolean checkStd)
 
         // Numeric reference?
         if (c == '#') {
-            c = buf[ptr++];
             int value = 0;
+            int pairValue = 0;
             int inputLen = mInputEnd;
-            if (c == 'x') { // hex
-                while (ptr < inputLen) {
+            final StringBuffer buffer = new StringBuffer(new String(buf));
```

Comment:
Ouch. This is not great; it'll create 8kB String just to parser a short entity.

Reply:
Actual n/m, I was able to remote it without ill effects it seems (was not being used -- unless it's something untested).